server/util/db/db.go

```diff
@@ -650,17 +651,41 @@ func GetConfiguredDatabase(env environment.Env) (interfaces.DBHandle, error) {
 	}
 	go statsRecorder.poll()
 
-	if *autoMigrateDBAndExit {
-		if err := runMigrations(driverName, primaryDB); err != nil {
-			log.Fatalf("Database auto-migration failed: %s", err)
+	if *autoMigrateDB || *autoMigrateDBAndExit || *printSchemaChangesAndExit {
+		sqlStrings := make([]string, 0)
+		if *printSchemaChangesAndExit {
+			if err := primaryDB.Callback().Raw().Register("save_sql", func(db *gorm.DB) {
+				executedSql := db.Statement.SQL.String()
+				if !strings.HasPrefix(executedSql, "SELECT") {
+					sqlStrings = append(sqlStrings, executedSql)
+				}
+			}); err != nil {
+				return nil, err
+			}
 		}
-		log.Infof("Database migration completed. Exiting due to --auto_migrate_db_and_exit.")
-		os.Exit(0)
-	}
-	if *autoMigrateDB {
+
 		if err := runMigrations(driverName, primaryDB); err != nil {
 			return nil, err
 		}
+
+		if *printSchemaChangesAndExit {
+			// Print schema changes to stdout
+			// Logs go to stderr, so this output will be easy to isolate and parse
+			if len(sqlStrings) > 0 {
+				fmt.Println("Auto-migration schema changes:")
```

Comment:
you could just remove this (and below) if you want the stdout to *only* be schema changes or nothing if there are none?